app/Models/AccessPoint.php

```diff
@@ -2,8 +2,37 @@
 
 namespace App\Models;
 
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+
 class AccessPoint extends DeviceRelatedModel
 {
-    protected $primaryKey = 'accesspoint_id';
+    use HasFactory;
+
+    public $primaryKey = 'accesspoint_id';
     public $timestamps = false;
+
+    protected $fillable = [
+        'device_id',
+        'name',
+        'radio_number',
+        'type',
+        'mac_addr',
+        'deleted',
+        'channel',
+        'txpow',
+        'radioutil',
+        'numasoclients',
+        'nummonclients',
+        'numactbssid',
+        'nummonbssid',
+        'interference',
+    ];
+
+    // ---- Define Relationships ----
+
+    public function device()
+    {
+        return $this->belongsTo(\App\Models\Devices::class, 'device_id');
```

Comment:
The model is call Device (not Devices). Also, you can omit this relationship, it is already added by DeviceRelatedModel

Reply:
removed